analysers/analyser_merge_milestone_FR_metropole.py

```diff
@@ -0,0 +1,103 @@
+            title = T_('Milestone integration suggestion'), **doc)
+        self.def_class_update_official(item = 8432, id = 44, level = 3, tags = ['merge', 'highway'],
+            title = T_('Milestone update'), **doc)
+
+        self.init(
+            "https://www.data.gouv.fr/fr/datasets/bornage-du-reseau-routier-national/",
+            "Bornage du réseau routier national",
+            CSV(Source(attribution ="data.gouv.fr:Ministère de la Transition écologique et solidaire", millesime = "01/2019",
+                    fileUrl = "https://www.data.gouv.fr/fr/datasets/r/fbc8b73b-a65c-486b-a710-ed22b9e4070c"),
+                    separator = "\t"),
+            Load("x", "y", srid = 2154,
+                xFunction = self.float_comma,
+                yFunction = self.float_comma,
+                where = lambda row: self.is_natref(row)),
+            Mapping(
+                select = Select(
+                    types = ["nodes"],
+                    tags = [{"highway": "milestone"}]),
+                osmRef = "nat_ref",
+                extraJoin = "ref",
+                conflationDistance = 150,
+                generate = Generate(
+                    static1 = {"highway": "milestone"},
+                    static2 = {"source": self.source},
+                    mapping1 = {
+                        "distance": 'pr',
+                        "nat_ref": lambda row: self.transform_to_natref(row),
+                        "ref": lambda row: self.transform_route(row['route']) }
+                    )))
+
+    def is_natref(self,row):
+        if len(row['depPr']) == 3:
+            return False
+        elif row['route'][0:1] == 'P':
+            #P for temporary,
+            return False
+        elif row['route'][0:2] in ('N1', 'N2'):
+            #N1 for future up_class and N2 for down_class road,
+            return False
+        elif row['route'][2:4] in ('A9', 'N9'):
+            #In metropole, it's not milestone but way_link or roundabout
+            return False
+        else:
+            return True
+
+    def transform_to_natref(self, row):
+        #dept must be on 2 caracter
+        dept = row['depPr']
+        if len(dept) == 1:
+            dept = '0' + dept
+
+        #C or '', not 'N'
+        concede = 'C' if row['concessionPr'] == 'C' else ''
+
+        #I is for ignore, sens is D,G or U for droite (sens croissant), gauche (sens décroissant), unique.
+        sens = row['cote']
+        if sens == 'I':
+            sens = 'U'
+
+        return dept + 'PR' + row['pr'] + sens + concede
+
+    def transform_route(self, route):
+        #remove multiple 0 and add space
+        if route[0:4] in ('A000', 'N000'): return route[0:1] + " " + route[4:]
+        elif route[0:3] in ('A00', 'N00'): return route[0:1] + " " + route[3:]
+        elif route[0:2] in ('A0', 'N0'): return route[0:1] + " " + route[2:]
+        else: return route[0:1] + " " + route[1:]
```

Comment:
It produces ref like `3 3A963010`, it is ok?

Reply:
No,
the csv contains PLO ("point localisant") which are milestone (for "section courante") or the first node for way_link/roundabout. The route fields contains string like 
A0010 or  33A963010. 

if route == 33A963010, it's a way link or roundabout because route[2:4]= A9 ;
must be filtered by : https://github.com/didier2020/osmose-backend/blob/rrn_milestone/analysers/analyser_merge_milestone_FR_metropole.py#L76

if route  == A0010
result is like 33PRxxD where xx is field pr

Reply:
It still produces ref like "3 3A901030CD".